### DISCUSSION TITLE
How to exit a fluentd plugin completely

### DISCUSSION THREAD

**Discussion by senalkularatne** [fluent/fluentd#3989] [Q&A]
What is the best way to exit a fluentd plugin for debugging purposes.

For example I've tried something like:

```
log.debug("Some message)
exit 1
```

However the process keeps going. How can I bring it to an end completely via code? Thanks.

**Comment by fujimotos** [comment]
> However the process keeps going. How can I bring it to an end completely via code? 

No this is not possible. Plugins are not supposed to exit. 

Fluentd just automatically restarts your plugin every time
it (self-)terminates.
